### DISCUSSION TITLE
Canvas rendered incorrectly in case a big self size

### DISCUSSION THREAD

**Discussion by stsiushkevich** [fabricjs/fabric.js#8072] [Q&A] [answered]
Version ```5.2.1```

Hello! In case not huge height, canvas renders my Rects properly:
![image](https://user-images.githubusercontent.com/12151060/179704899-9c72a393-fb4c-4296-9147-b9da1fcd5bdb.png)
![image](https://user-images.githubusercontent.com/12151060/179704926-79a448a4-2338-4ebf-b3d4-ea845ab1911b.png)

But when I set a huge heigth, my Rects renderen incorrectly:
![image](https://user-images.githubusercontent.com/12151060/179706030-707c89a3-9754-47c1-ab61-139b39633761.png)
![image](https://user-images.githubusercontent.com/12151060/179706051-68c39175-b210-426b-9fe9-15a037f245c6.png)

Rects are rendered but not visible! I can click them and handle click events. Methods ```renderAll``` ```requestRenderAll``` are not working in this case.

**Comment by ShaMan123** [comment, marked as answer]
It seems like a problem on your side.
And it is bad practice to use a huge canvas, lots of memory use. You should use `viewportTransform`

**Reply by stsiushkevich** [reply]
> It seems like a problem on your side. And it is bad practice to use a huge canvas, lots of memory use. You should use `viewportTransform`

Thank you! I splitted a single canvas on the smaller ant thay're working properly.
